### ISSUE TITLE
Only the first terminal window adheres to tiling, succeeding windows do not

### ISSUE THREAD

**Issue by tina-nova** [xmonad/xmonad#185]
### Problem Description

It seems that a recent update has created a problem with tiling my terminal windows, whether by pressing the mod+shift+enter hotkey or by opening a new terminal window with ctrl+shift+n.

I thought maybe it was a bug in xfce4-terminal so I tried both lxterminal and gnome-terminal, ending up with similar results. Other programs seem fine; it looks like the only issue is with opening new windows from any terminal program after the first window is opened.

I use ArchLinux with xmonad + xmonad-contrib, version 0.15-2. The problem also exists in the AUR packages xmonad-git + xmonad-contrib-git, which I'm assuming are as bleeding-edge as it gets outside of xmonad-testing.  Will provide more details if needed.

### Configuration File

I changed nothing with my configuration except when alternating terminals, but here's my config anyway:

> ```
> import XMonad
> import qualified Data.Map as M
> import qualified XMonad.StackSet as W
> import XMonad.Actions.CycleWS
> import XMonad.Config.Desktop
> import XMonad.Config.Xfce
> import XMonad.Layout.Spacing
> import XMonad.Hooks.ManageDocks
> import XMonad.Hooks.ManageHelpers 
> import XMonad.Layout.NoBorders
> import XMonad.Hooks.EwmhDesktops
> import XMonad.Util.EZConfig
> import XMonad.Actions.GridSelect
> import XMonad.Layout.PerWorkspace
> import XMonad.Layout.GridVariants
> import XMonad.Layout.ComboP
> import XMonad.Layout.Reflect
> import XMonad.Layout.TwoPane
> import XMonad.Layout.Dishes
> import XMonad.Hooks.SetWMName
> import XMonad.Util.CustomKeys
> import XMonad.Util.EZConfig
> 
> 
> myLayoutHook = 	modWorkspaces ["1","2","3","4","5","6","7","8","9","F1:WebApps","F2:Base","F3:Monitoring","F4:Virtualbox","F5:Audio","F6:Spotify","F7:VNC","F8:Deluge","F9:Skype","F10:Viber","F11:Fullscreen","F12:Steam"] avoidStruts (onWorkspace "F3:Monitoring" d a ||| b ||| c)
> 		where 
> 			d = combineTwoP (Mirror (Tall 1 0.03 0.66)) (Full) (combineTwoP (reflectHoriz $ Tall 1 0.03 0.5) (Full) (combineTwoP (Dishes 1 0.5) (Full) (Full) (ClassName "ConkyNet") ) (ClassName "ConkyPC")) (ClassName "ConkyHDD")
> 			a = Tall 1 (3/100) (1/2) 
> 			b = Mirror (a)
> 			c = Full
> 
> myTerminal = "lxterminal"
> 
> myWorkspaces = ["1","2","3","4","5","6","7","8","9","F1:WebApps","F2:Base","F3:Monitoring","F4:Virtualbox","F5:Audio","F6:Spotify","F7:VNC","F8:Deluge","F9:Skype","F10:Viber","F11:Fullscreen","F12:Steam"]
> 
> myKeys =
> 	[
> 	((0, xK_F13), goToSelected defaultGSConfig)
> 	] ++ [
> 	((mod4Mask, key), (windows $ W.greedyView ws))
> 		| (key,ws) <- zip workspaceKeys myWorkspaces 
> 	] ++ [
> 	((mod4Mask .|. shiftMask, key), (windows $ W.shift ws))
> 		| (key,ws) <- zip workspaceKeys myWorkspaces 
> 	] ++ [
> 	((mod4Mask, xK_h), sendMessage Shrink)
> 	] ++ [
> 	((mod4Mask, xK_l), sendMessage Expand)
> 	] ++ [
> 	((mod4Mask .|. shiftMask, xK_o), shiftNextScreen)
> 	] ++ [ 
> 	((mod4Mask .|. shiftMask, xK_Tab), nextScreen)
> 	]
> 	where 	
> 		workspaceKeys = [xK_1 .. xK_9] ++ [xK_F1 .. xK_F12]
> 
> main = xmonad $ ewmh (xfceConfig
> 		{modMask = mod4Mask
> 		,startupHook = setWMName "LG3D"
> 		,terminal = myTerminal 
> 		
> 		,manageHook = ( isFullscreen --> doFullFloat ) <+> manageHook defaultConfig <+> manageDocks <+> myManageHook
> 		,borderWidth = 0
> 		,handleEventHook = fullscreenEventHook <+> docksEventHook
> 		,workspaces = myWorkspaces
> 		,layoutHook = myLayoutHook
> 		} `additionalKeys` (myKeys))
> 
> myManageHook = composeAll . concat $
>   [ [ resource =? name --> doIgnore | name <- ignore ]
>   , [ title =? name --> doIgnore | name <- ignore ]
>   , [ className =? name --> doIgnore | name <- ignore ]
>   , [ resource =? name --> doFloat | name <- floaters ]
>   , [ title =? name --> doFloat | name <- floaters ]
>   , [ className =? name --> doFloat | name <- floaters ]
>   , [ composeOne [ isFullscreen -?> doFullFloat] ] --full float fullscreen flash
>   , [ isDialog --> doCenterFloat] --full float fullscreen flash
>   , [ --className =? "Steam" --> doFloat,
>       className =? "steam" --> doFullFloat, -- bigpicture-mode
>       {-className =? "MainThrd" --> doFloat,-}
>      -- className =? "Steam" --> doFloat,
>       title =? "plasma-desktop" --> doIgnore
>   ]
>   , [ className =? "Seamonkey" <&&> resource =? "Navigator" --> doShift "F1:WebApps" ]
>   , [ className =? "Spotify" --> doShift "F6:Spotify" ]
>   , [ className =? "banshee" --> doShift "F5:Audio" ]
>   , [ className =? "Steam" --> doShift "F12:Steam" ] 
>   , [ className =? "steam" --> doShift "F12:Steam" ]
>   , [ className =? "Deluge" --> doShift "F8:Deluge" ]
>   , [ className =? "Skype" --> doShift "F9:Skype" ]
>   , [ className =? "ViberPC" --> doShift "F10:Viber" ]
>   , [ className =? "libreoffice-base" <&&> resource =? "libreoffice" --> doShift "F2:Base" ]
>   , [ className =? "VirtualBox" --> doShift "F4:Virtualbox" ]
>   , [ className =? "Conky" --> doShift "F3:Monitoring" ]
>   , [ className =? "ConkyHDD" --> doShift "F3:Monitoring" ]
>   , [ className =? "ConkyNet" --> doShift "F3:Monitoring" ]
>   , [ className =? "ConkyPC" --> doShift "F3:Monitoring" ]
>   , [ className =? "Pavucontrol" --> doShift "F5:Audio" ]
>   , [ className =? "Remmina" --> doShift "F7:VNC" ]
>   , [ className =? "xfce4-panel" --> doIgnore ]
>   ]
>   where
>     floaters = [""]--"glut", "MPlayer", "Vlc", "Gimp", "Xfce4-appfinder", "Xfrun4"]
>     ignore = ["desktop_window", "kdesktop"]
> ```

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md) 

  - [ ] ~~I tested my configuration with [xmonad-testing](https://github.com/xmonad/xmonad-testing)~~ I'm a pathetic archlinux user who tried to compile this on my own because I couldn't find it on the AUR. It nevertheless failed to compile, so I gave up.

**Comment by chrissound** [comment]
You could possibly try a much simpler layout and see if you hit the same issues?

**Comment by liskin** [comment]
@budomino What do you mean by "doesn't adhere to tiling"? Can you attach a screenshot or a video?

**Comment by tina-nova** [comment]
Hi, so I figured it out. The cause of the issue is lxterminal being detected as a floater as per the second to the last line. Apparently the lxterminal window starts with a blank title and that is probably what xmonad detects. I just commented out all references to floaters as I don't use floaters anymore.

For future reference, I deleted these lines:


>   , [ resource =? name --> doFloat | name <- floaters ]
>   , [ title =? name --> doFloat | name <- floaters ]
>   , [ className =? name --> doFloat | name <- floaters ]

...

>    floaters = [""]--"glut", "MPlayer", "Vlc", "Gimp", "Xfce4-appfinder", "Xfrun4"]

For clarity, the real issue was a simple logic error on my part. All lxterminal windows launched via hotkey would be considered floating and thus anchored to the upper left corner of the screen as a small window.
